code/game/machinery/bots/medbot.dm

```diff
@@ -472,47 +472,54 @@
 			spawn(INJECTION_TIME)
 				inject_patient()
 
-//	src.speak(reagent_id)
-	reagent_id = null
-	return
-
 /obj/machinery/bot/medbot/proc/inject_patient()
-	if ((get_dist(src, src.patient) <= 1) && (src.on))
-		if((reagent_id == "internal_beaker") && (src.reagent_glass) && (src.reagent_glass.reagents.total_volume))
-			src.reagent_glass.reagents.trans_to(src.patient,src.injection_amount) //Inject from beaker instead.
-			src.reagent_glass.reagents.reaction(src.patient, 2)
+	if ((get_dist(src, patient) <= 1) && (on))
+		if((reagent_id == "internal_beaker") && (reagent_glass) && (reagent_glass.reagents.total_volume))
+			reagent_glass.reagents.trans_to(patient,injection_amount) //Inject from beaker instead.
+			reagent_glass.reagents.reaction(patient, 2)
 		else
-			src.patient.reagents.add_reagent(reagent_id,src.injection_amount)
-		visible_message("<span class='danger'>[src] injects [src.patient] with the syringe!</span>")
+			patient.reagents.add_reagent(reagent_id,injection_amount)
+
+		visible_message("<span class='danger'>[src] injects [patient] with the syringe!</span>")
 
-	src.icon_state = "[src.icon_initial][src.on]"
-	src.currently_healing = 0
+	icon_state = "[icon_initial][on]"
+	currently_healing = 0
+	reagent_id = null
 	return
 
 /obj/machinery/bot/medbot/proc/speak(var/message)
-	if((!src.on) || (!message))
+	if((!on) || (!message))
 		return
 	visible_message("<b>[src]</b> beeps, \"[message]\"",\
 		drugged_message="<b>[src]</b> beeps, \"[pick("FEED ME HUMANS","LET THE BLOOD FLOW","BLOOD FOR THE BLOOD GOD","I SPREAD DEATH AND DESTRUCTION","EXTERMINATE","I HATE YOU!","SURRENDER TO YOUR MACHINE OVERLORDS","FEED ME SHITTERS")]\"")
 	return
 
 /obj/machinery/bot/medbot/bullet_act(var/obj/item/projectile/Proj)
 	if(Proj.flag == "taser")
-		src.stunned = min(stunned+10,20)
+		stunned = min(stunned+10,20)
 	..()
 
 /obj/machinery/bot/medbot/explode()
-	src.on = 0
+	on = 0
 	visible_message("<span class='danger'>[src] blows apart!</span>", 1)
 	var/turf/Tsec = get_turf(src)
 
-	new /obj/item/weapon/storage/firstaid(Tsec)
+	switch(skin)
```

Comment:
Primary change two - Upon medbot destruction, it switches through skins much like it does on new, where it then drops the corresponding medkit upon destruction, defaulting to the basic medkit